jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SniX509ExtendedKeyManager.java

```diff
@@ -81,29 +88,51 @@ protected String chooseServerAlias(String keyType, Principal[] issuers, Collecti
                 {
                     SslContextFactory.AliasSNIMatcher matcher = (SslContextFactory.AliasSNIMatcher)m;
                     host = matcher.getHost();
-                    x509 = matcher.getX509();
                     break;
                 }
             }
         }
 
-        if (LOG.isDebugEnabled())
-            LOG.debug("Matched {} with {} from {}", host, x509, Arrays.asList(aliases));
+        SslContextFactory.SNISelector sniSelector = _sslContextFactory.getSNISelector();
+        if (sniSelector == null)
+            sniSelector = _sslContextFactory;
 
-        // Check if the SNI selected alias is allowable
-        if (x509 != null)
+        try
         {
-            for (String a : aliases)
+            // Filter the certificates by alias.
+            Collection<X509> certificates = new ArrayList<>();
```

Comment:
Perhaps use streams API as that is the style mostly being used elsewhere in this code now:
```java
   aliases.stream()
       .map(_sslContextFactory::getX509)
       .filter(x->x!=null) // sure there is a better way
       .collect ( whatever );
```